hack/setup.sh

```diff
@@ -0,0 +1,129 @@
+
+echo "Configuring Knative Serving to use Kourier..."
+kubectl patch configmap/config-network \
+  --namespace knative-serving \
+  --type merge \
+  --patch '{"data":{"ingress-class":"kourier.ingress.networking.knative.dev"}}'
+
+echo "Fetching External IP or CNAME..."
+kubectl --namespace kourier-system get service kourier
+
+echo "Waiting for all Knative Serving pods to be in the Running state..."
+
+while IFS= read -r line; do
+    POD_NAME=$(echo $line | awk '{print $1}')
+    POD_STATUS=$(echo $line | awk '{print $3}')
+    if [ "$POD_STATUS" != "Running" ]; then
+        echo "Waiting for pod $POD_NAME to be in Running state..."
+        sleep 5
+        kubectl get pods --namespace knative-serving | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods --namespace knative-serving | grep -v NAME)
+
+echo "All pods are running."
+
+echo "Verifying Knative Serving Installation..."
+kubectl get pods --namespace knative-serving
+
+echo "<Success!> Knative Serving Installation Completed Successfully!"
+
+
+
+echo "Installing Knative Eventing Custom Resource Definitions..."
+kubectl apply -f https://github.com/knative/eventing/releases/download/knative-v1.12.4/eventing-crds.yaml
+
+echo "Installing Knative Eventing Core Components..."
+kubectl apply -f https://github.com/knative/eventing/releases/download/knative-v1.12.4/eventing-core.yaml
+
+echo "Waiting for all Knative Eventing pods to be in the Running or Completed state..."
+
+while IFS= read -r line; do
+    POD_NAME=$(echo $line | awk '{print $1}')
+    POD_STATUS=$(echo $line | awk '{print $3}')
+    if [ "$POD_STATUS" != "Running" ] && [ "$POD_STATUS" != "Completed" ]; then
+        echo "Waiting for pod $POD_NAME to reach Running or Completed state..."
+        sleep 5
+        kubectl get pods -n knative-eventing | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods -n knative-eventing | grep -v NAME)
+
+echo "All Knative Eventing pods are in the desired state."
+
+echo "<Success!>Knative Eventing Installation Completed Successfully!"
+
+
+echo "Before we proceed with installing the config, we need to check whether ko is installed or not."
+
+if command -v ko >/dev/null 2>&1; then
+    echo "ko is installed, proceeding with the installation of the config."
+else
+    echo "ko is not installed. Installing ko..."
+    # Check if Go is installed
+    if ! command -v go >/dev/null 2>&1; then
+        echo "Go is not installed. Please manually install Go first."
+        exit 1
+    fi
+
+    # Install ko
+    echo "Installing ko..."
+    GO111MODULE=on go get github.com/google/ko/cmd/ko
+
+    # Verifying ko installation
+    if command -v ko >/dev/null 2>&1; then
+        echo "ko successfully installed. Proceeding with the installation of the config"
+    else
+        echo "Failed to install ko. Please try to manually install ko."
+        exit 1
+    fi
+fi
```

Comment:
I don't think installing ko should be a part of this script. We should just let users know to install ko themselves

Reply:
Sounds good. Instead I will prompt user that ko is not installed, and ask them to install manually.